### PR TITLE
fix(platform): contested username time fix

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
Contested usernames were still set as 2 weeks. This PR fixes that.

## What was done?
Properly set network for Drive Config

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [X] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive-abci/src/config.rs [hunk 1]
```diff
@@ -877,4 +958,36 @@ mod tests {
...
+        assert_eq!("/tmp/rejected", env::var("REJECTIONS_PATH").unwrap());
+
+        let config = super::PlatformConfig::from_env().expect("expected config from env");
+        assert!(config.execution.verify_sum_trees);
+        assert_eq!(config.validator_set.quorum_type, QuorumType::Llmq25_67);
+        assert_eq!(config.network, config.drive.network);
+        assert_eq!(config.network, Network::Testnet);
+        for id in vectors {
+            matches!(config.abci.log[id.0].destination, LogDestination::Bytes);
+        }
```
Comment (shumkov) [inline packages/rs-drive-abci/src/config.rs:991]: Not need

File: packages/rs-drive-abci/src/config.rs [hunk 2]
```diff
@@ -877,4 +958,36 @@ mod tests {
...
+
+        let envfile = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR")).join(".env.testnet");
+
+        dotenvy::from_path(envfile.as_path()).expect("cannot load .env file");
+        assert_eq!("/tmp/db", env::var("DB_PATH").unwrap());
+        assert_eq!("/tmp/rejected", env::var("REJECTIONS_PATH").unwrap());
+
+        let config = super::PlatformConfig::from_env().expect("expected config from env");
+        assert!(config.execution.verify_sum_trees);
+        assert_eq!(config.validator_set.quorum_type, QuorumType::Llmq25_67);
```
Comment (shumkov) [inline packages/rs-drive-abci/src/config.rs:986]: Not need

File: packages/rs-drive/src/config.rs [hunk 3]
```diff
@@ -118,10 +118,7 @@ pub struct DriveConfig {
     /// The network type
     #[cfg_attr(
         feature = "serde",
-        serde(
-            default = "DriveConfig::default_network",
-            deserialize_with = "from_str_to_network_with_aliases"
```
Comment (shumkov) [inline packages/rs-drive/src/config.rs:123]: Remove function below

File: packages/rs-platform-version/src/version/v2.rs [hunk 4]
```diff
@@ -1247,7 +1247,7 @@ pub const PLATFORM_V2: PlatformVersion = PlatformVersion {
         },
         voting_versions: VotingVersions {
             default_vote_poll_time_duration_mainnet_ms: 1_209_600_000, //2 weeks
-            default_vote_poll_time_duration_test_network_ms: 5_400_000, //90 minutes
+            default_vote_poll_time_duration_test_network_ms: 1_209_600_000, //2 weeks
```
Comment (shumkov) [inline packages/rs-platform-version/src/version/v2.rs:1250]: You bring it back to keep compatibility with testnet data?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-platform-version/src/version/v2.rs (hunk 4 above)
Comment (QuantumExplorer) [reply packages/rs-platform-version/src/version/v2.rs:1250]: yes

File: packages/rs-drive/src/config.rs (hunk 3 above)
Comment (QuantumExplorer) [reply packages/rs-drive/src/config.rs:123]: done

File: packages/rs-drive-abci/src/config.rs (hunk 2 above)
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/config.rs:986]: why not, we want to see that we are on testnet.

File: packages/rs-drive-abci/src/config.rs [hunk 5]
```diff
@@ -877,4 +958,36 @@ mod tests {
...
+            ("STDOUT", "pretty"),
+            ("UPPERCASE", "json"),
+            ("lowercase", "pretty"),
+            ("miXedC4s3", "full"),
+            ("123", "compact"),
+        ];
+        for vector in vectors {
+            env::set_var(format!("ABCI_LOG_{}_DESTINATION", vector.0), "bytes");
+            env::set_var(format!("ABCI_LOG_{}_FORMAT", vector.0), vector.1);
+        }
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/config.rs]: removed.

File: packages/rs-drive-abci/src/config.rs [hunk 6]
```diff
@@ -216,6 +216,86 @@ pub struct PlatformConfig {
     pub tokio_console_retention_secs: u64,
 }
 
+// Define an intermediate struct that mirrors PlatformConfig
+#[derive(Deserialize)]
+struct PlatformConfigIntermediate {
+    /// The network type
+    #[serde(
+        default = "PlatformConfig::default_network",
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/config.rs:224]: no, we need a default or else it won't deserialize.

File: packages/rs-drive-abci/src/config.rs [hunk 7]
```diff
@@ -877,4 +958,36 @@ mod tests {
...
+        for vector in vectors {
+            env::set_var(format!("ABCI_LOG_{}_DESTINATION", vector.0), "bytes");
+            env::set_var(format!("ABCI_LOG_{}_FORMAT", vector.0), vector.1);
+        }
+
+        let envfile = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR")).join(".env.testnet");
+
+        dotenvy::from_path(envfile.as_path()).expect("cannot load .env file");
+        assert_eq!("/tmp/db", env::var("DB_PATH").unwrap());
+        assert_eq!("/tmp/rejected", env::var("REJECTIONS_PATH").unwrap());
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/config.rs]: removed.

File: packages/rs-drive-abci/src/config.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/config.rs]: removed.

File: packages/rs-drive-abci/src/config.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/config.rs:224]: Yeah, but it very fragile. I would make it strict. But probably it's a topic for another PR

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/src/config.rs [hunk 8]
```diff
@@ -129,7 +129,7 @@ pub struct ExecutionConfig {
 /// ``
 ///
 /// [`verify_sum_trees`]: PlatformConfig::verify_sum_trees
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize)]
 // NOTE: in renames, we use lower_snake_case, because uppercase does not work; see
 // https://github.com/softprops/envy/issues/61 and https://github.com/softprops/envy/pull/69
 pub struct PlatformConfig {
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/config.rs:135]: done

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**